src/element/resizeElements.ts

```diff
@@ -694,62 +728,71 @@ const resizeMultipleElements = (
       height: number;
       x: number;
       y: number;
+      angle: number;
       points?: Point[];
       fontSize?: number;
       baseline?: number;
+      scale?: [number, number];
     } = {
       width,
       height,
       x,
       y,
+      angle,
       ...rescaledPoints,
     };
 
-    let boundTextUpdates: { fontSize: number; baseline: number } | null = null;
+    if (isImageElement(element) && targetElements.length === 1) {
+      update.scale = [
+        element.scale[0] * flipFactorX,
+        element.scale[1] * flipFactorY,
+      ];
+    }
 
-    const boundTextElement = getBoundTextElement(element.latest);
+    if (isTextElement(element)) {
+      const metrics = measureFontSizeFromWidth(element, width, height);
+      update.fontSize = metrics?.size ?? element.fontSize;
+      update.baseline = metrics?.baseline ?? element.baseline;
+    }
 
-    if (boundTextElement || isTextElement(element.orig)) {
-      const updatedElement = {
-        ...element.latest,
-        width,
-        height,
-      };
-      const metrics = measureFontSizeFromWidth(
-        boundTextElement ?? (element.orig as ExcalidrawTextElement),
-        boundTextElement
-          ? getBoundTextMaxWidth(updatedElement)
-          : updatedElement.width,
-        boundTextElement
-          ? getBoundTextMaxHeight(updatedElement, boundTextElement)
-          : updatedElement.height,
+    // TODO remove this after solving the issue with changing bounds of linear
+    // elements with roughness > 0
+    if (isLinearElement(element) && (isFlippedByX || isFlippedByY)) {
+      const origBounds = getElementPointsCoords(element, element.points);
+      const newBounds = getElementPointsCoords(
+        { ...element, x, y },
+        rescaledPoints.points!,
       );
+      const origXY = [element.x, element.y];
+      const newXY = [x, y];
+
+      const calculateCorrenction = (axis: "x" | "y") => {
```

Comment:
Currently linear elements with roughness > 0 change their size after flipping, so if you flip such an element multiple times it can slowly drift away from the original position. This code aligns the flipped element with its original center.